qmlti/lti/LTI_Tool_Provider.php

```diff
@@ -209,11 +208,16 @@ public function execute() {
     if (isset($_POST['launch_presentation_return_url'])) {
       $this->return_url = $_POST['launch_presentation_return_url'];
     }
+
 #
 ### Perform action
 #
     if ($this->authenticate()) {
       $this->doCallback();
+      error_log("Callback successful.");
+    } else {
+      error_log( "Error authenticating." );
```

Comment:
Does this write a line per log message? if so you will want to joins these 2 lines into one to aid readability. Imagine if there were 10,000 users, these 2 lines could be some distance apart.